docs/development/design/shared-cidr.md

```diff
@@ -0,0 +1,13 @@
+# Why we should be using a single CG-NAT IPv4 Address space in Nexodus
```

Comment:
Would you mind tweaking this to follow the template?

https://docs.nexodus.io/development/design/TEMPLATE/

Some of the "alternatives considered" are discussed in PR comments.